src/pages/self-hosting/railway.mdx

```diff
@@ -0,0 +1,187 @@
+
+[![Railway deployment template](/assets/images/self-hosting/railway/railway-template.png)](https://railway.app/template/FgdM-Z?referralCode=rG9hj-)
+
+## Prerequisites
+- A Railway account on the [Hobby](https://railway.app/pricing) tier or better
+- A [Google](/access-control/authentication/oauth-sso#google), [GitHub](/access-control/authentication/oauth-sso#github), or [GitLab](/access-control/authentication/oauth-sso#gitlab) OAuth SSO for authentication
+
+## Known Issues
+
+- **Nginx routing issues**: Redeployment of services (frontend, backend, worker), especially due to configuration changes (environment variables, settings, adding custom domains, etc.), can cause the nginx routing to time out due to IPv6 changes after deployment. If you encounter such issues, please redeploy the `frontend`, `backend` and `railway-nginx` services respectively. Make sure to redeploy the nginx service **last** after all other services have been finalized and re-deployed.
+
+- **Secret references**: To make configuration easier, the Phase Console Railway deployment template uses numerous secret and variable references. These references are resolved only during a service deployment. Future configuration changes to other services may not reflect immediately until a redeployment is triggered (as mentioned in the previous point).
+
+For example, the Phase `backend` service references an environment variable from the `railway-nginx` service. If a change occurs in the referenced value `${{railway-nginx.RAILWAY_PUBLIC_DOMAIN}}` (e.g., because a new custom domain was added), the `backend` and the `railway-nginx` service needs to be redeployed respectively:
+
+```fish
+ALLOWED_HOSTS=healthcheck.railway.app,${{RAILWAY_PRIVATE_DOMAIN}},${{railway-nginx.RAILWAY_PUBLIC_DOMAIN}}
+```
+
+## Installation Steps
+
+### 1. Configure the Railway Deployment Template
+
+1. Go to the [Phase Console](https://railway.app/template/FgdM-Z?referralCode=rG9hj-) Railway deployment template page and click **Deploy Now**.
+
+![Railway template click deploy now button](/assets/images/self-hosting/railway/configure/1-railway-template-deploy-now.png)
+
+2. Click **Configure** on each service container listed, fill out the required Environment variables, and click **Save Config**.
+
+![Configure environment variables](/assets/images/self-hosting/railway/configure/2-railway-template-services-configure.png)
+
+Important notes:
+- Read the descriptions for each Environment variable as they contain setup instructions
+- For security and consistency, we do not autogenerate secrets using Railway's secret function syntax. Generate them manually using `openssl` like this: `openssl rand -hex 32`
+- Some Environment variables need to be set across multiple services
+- You'll need to set up a Google, GitHub, or GitLab OAuth Application and provide their credentials as Environment variables. This can be done after setting up a custom domain in [Step 2](/self-hosting/railway#2-deploy-the-template).
```

Comment:
```suggestion
- You'll need to set up a Google, GitHub, or GitLab OAuth Application and provide their credentials as environment variables. This can be done after setting up a custom domain in [Step 2](/self-hosting/railway#2-deploy-the-template).
```